Extensions/xkit_patches.js

```diff
@@ -549,6 +565,43 @@ XKit.extensions.xkit_patches = new Object({
 				}
 			};
 
+			XKit.css_map = {
+				callbacks: {},
+				done: false,
+				cssMap: null,
+				add_callback: function(extension, func) {
+					if (this.done) {
+						func.call(XKit.extensions[extension], this.cssMap);
+					} else {
+						this.callbacks[extension] = func;
+					}
+				},
+				setCssMap: function(cssMap) {
+					this.done = true;
+					this.cssMap = cssMap;
+
+					const callbacks = this.callbacks;
+					for (const extension in callbacks) {
+						callbacks[extension].call(XKit.extensions[extension], this.cssMap);
+					}
+				},
+				keyToCss: function(key) {
+					if (!this.cssMap || !this.cssMap.hasOwnProperty(key)) {
+						return;
+					}
+					return this.cssMap[key].map(cls => '.' + cls).join(', ');
```

Comment:
unfortunately without [`:is()`/`:matches()`/`:any()`](https://developer.mozilla.org/en-US/docs/Web/CSS/:is#Browser_compatibility) we're gonna have to do something very different here (unless we want to do all of our styling via jQuery). i think the simplest solution would be for this function to take the parent selector as a second argument, so for example
```js
XKit.css_map.keyToCss('post', '.xblacklist_blacklisted_post')
```
would return
```css
.xblacklist_blacklisted_post ._2OqKn,
.xblacklist_blacklisted_post ._2-mU3,
.xblacklist_blacklisted_post ._1yQrs,
.xblacklist_blacklisted_post ._1Wz4U
```

Reply:
i feel like this still won't solve all our problems though. the css map is useful, but being so unspecific is gonna trip us over easily. we need more than classes to target specific elements now

Reply:
I ended up writing an ad hoc `map`+`join` since it's just in the one place. It could definitely be added as a proper variant of keyToCss though but I'm not sure what the best API would be. I think only supporting parent selectors is a bit too narrow but parent+children should just about cover 90% of use cases